Move qualcomm.passes to qualcomm._passes

tarun292 opened the pull request:
Differential Revision: D63926845

chiwwang commented:
@haowhsu-quic  @shewu-quic  @chunit-quic  @winskuo-quic

shewu-quic reviewed (approved):
If possible, could you help to change [this line](https://github.com/pytorch/executorch/blob/ac2ae072e0c54171b995432129ea5dcd20d74eb0/backends/qualcomm/README.md?plain=1#L49) in README.md

tarun292 committed:
Move qualcomm.passes to qualcomm._passes (#5920)

Summary:
Changing qualcomm.passes to qualcomm._passes to indicate that these passes are not covered under the API stability guarantee.

Pull Request resolved: https://github.com/pytorch/executorch/pull/5920

Reviewed By: helunwencser

Differential Revision: D63926845

tarun292 commented:
@pytorchbot cherry-pick --onto release/0.4 -c critical
